pkg/apis/core/types_controllerdeployment.go

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ControllerDeployment contains information about how this controller is deployed.
+type ControllerDeployment struct {
+	metav1.TypeMeta
+	// Standard object metadata.
+	metav1.ObjectMeta
+	// Spec contains the specification of this deployment.
+	Spec ControllerDeploymentSpec
```

Comment:
I'm wondering whether we need `Spec` in this resource - WDYT?

Reply:
I was also pondering about that and eventually decided for the `Spec` because then we are open to add a `Status` in the future. Otherwise, I don't have a strong opinion. WDYT?

Reply:
Not sure, what would we need a status for? Isn't this resource similar to `ConfigMap`s/`ShootState`s/...?

Reply:
You're right, we will probably always maintain the status of an extension controller "deployment" in the `ControllerInstallation`. I removed the spec in my latest commit.